include-refactoring/xsimd/types/xsimd_utils.hpp

```diff
@@ -89,6 +89,18 @@ namespace xsimd
     template <class T>
     using as_unsigned_integer_t = typename as_unsigned_integer<T>::type;
 
+    /*********************
+     * as_signed_integer *
+     *********************/
+
+    template <class T>
+    struct as_signed_integer : std::make_signed<T>
```

Comment:
this could be a `using directive`

Reply:
Theoretically yes, but I wanted to keep it consistent with `as_unsigned_integer` and the pattern from the STL: use `struc traits_name`  and inheritance so that you can specialize it, and `using traits_name_t = typename traits_name<T>::type` for keepping the code concise.